web/libs/ui/src/lib/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,57 @@
+import { nanoid } from "nanoid";
+import React, { type InputHTMLAttributes } from "react";
+import styles from "./checkbox.module.scss";
+
+export interface CheckboxProps extends Omit<InputHTMLAttributes<HTMLInputElement>, "type"> {
+  indeterminate?: boolean;
+}
+
+export const Checkbox = ({ checked, indeterminate, style, onChange, children, ...props }: CheckboxProps) => {
+  const checkboxRef = React.createRef();
```

Comment:
This should be just using a standard ref hook.
```suggestion
  const checkboxRef = useRef();
```